neuralhydrology/modelzoo/transformer.py

```diff
@@ -0,0 +1,167 @@
+        The run configuration.
+    """
+
+    def __init__(self, cfg: Dict):
+        super(Transformer, self).__init__(cfg=cfg)
+
+        if cfg.embedding_hiddens:
+            LOGGER.warning("## Warning: Embedding settings are ignored. Use EmbCudaLSTM for embeddings")
+
+        input_size = len(cfg.dynamic_inputs + cfg.evolving_attributes + cfg.hydroatlas_attributes +
+                         cfg.static_attributes)
+        if cfg.use_basin_id_encoding:
+            input_size += cfg.number_of_basins
+
+        if cfg.head.lower() == "umal":
+            input_size += 1
+
+        # embedding 
+        self.embedding_dim = cfg.transformer_embedding_dimension
+        self.embedding = nn.Linear(in_features=input_size, 
+                                   out_features=self.embedding_dim)
+
+        # positional encoder
+        self.posistional_encoding_type = cfg.transformer_positional_encoding_type
+        if self.posistional_encoding_type.lower() in ['concat', 'cat', 'concatenate']:
+            self.encoder_dim = self.embedding_dim*2
+        elif self.posistional_encoding_type.lower() in ['sum', 'add', 'addition']:
+            self.encoder_dim = self.embedding_dim
+        else:
+            raise RuntimeError(f"Unrecognized positional encoding type: {self.posistional_encoding_type}")
+        self.pos_encoder = PositionalEncoding(d_model=self.embedding_dim, 
+                                              dropout=cfg.transformer_positional_dropout, 
+                                              max_len=cfg.seq_length)
+
+        # positional mask
+        self.mask = None
+
+        # encoder
+        encoder_layers = nn.TransformerEncoderLayer(d_model=self.encoder_dim, 
+                                                    nhead=cfg.transformer_nheads, 
+                                                    dim_feedforward=cfg.transformer_dim_feedforward, 
+                                                    dropout=cfg.transformer_dropout)
+        self.encoder = nn.TransformerEncoder(encoder_layer=encoder_layers, 
+                                             num_layers=cfg.transformer_nlayers,
+                                             norm=None) 
+
+        # head (instead of a decoder)
+        self.dropout = nn.Dropout(p=cfg.output_dropout)
+        self.head = get_head(cfg=cfg, n_in=self.encoder_dim, n_out=self.output_size)
+
+        # init weights and biases 
+        self._reset_parameters()
+
+    def _reset_parameters(self): 
+        initrange = 0.1
+        for layer in self.encoder.layers:
+            layer.linear1.weight.data.uniform_(-initrange, initrange)
+            layer.linear1.bias.data.zero_()
+            layer.linear2.weight.data.uniform_(-initrange, initrange)
+            layer.linear2.bias.data.zero_()
+        self.embedding.weight.data.uniform_(-initrange, initrange)
+        self.embedding.bias.data.zero_()
+
+    def forward(self, data: Dict[str, torch.Tensor], **kwargs) -> Dict[str, torch.Tensor]:
```

Comment:
Any particular reason for the `**kwargs`. We don't have them in any other model and I don't see a reason to have them here neither.
```suggestion
    def forward(self, data: Dict[str, torch.Tensor]) -> Dict[str, torch.Tensor]:
```